tla-types/src/main/scala/at/forsyte/apalache/tla/typecomp/subbuilder/SetBuilder.scala

```diff
@@ -44,21 +43,35 @@ trait SetBuilder extends UnsafeSetBuilder {
 
   /** { x \in set: p } */
   def filter(x: TBuilderInstruction, set: TBuilderInstruction, p: TBuilderInstruction): TBuilderInstruction =
-    for {
-      xEx <- x
-      setEx <- set
-      pEx <- p
-      _ <- markAsBound(xEx)
-    } yield _filter(xEx, setEx, pEx)
+    boundVarIntroductionTernary(_filter)(x, set, p)
 
   /** { mapExpr: x1 \in set1 , ..., xN \in setN }, must have at least 1 var-set pair */
   def map(
       e: TBuilderInstruction,
       varSetPairs: (TBuilderInstruction, TBuilderInstruction)*): TBuilderInstruction = {
-    val args = varSetPairs.flatMap { case (k, v) =>
-      Seq(k, v)
-    }
-    mapMixed(e, args: _*)
+    for {
+      mapExpr <- e
+      boundAfterMapExpr <- allBound // xi may not appear as bound in mapExpr
+      _ = {}
```

Comment:
What is this doing?

Reply:
Skillfully avoiding `fmt-check`. It's leftover code that should be rm'd.